pyiron_base/generic/units.py

```diff
@@ -0,0 +1,300 @@
+
+    The unit converter instance can then be used to obtain conversion factors between code and base units either as a
+    `pint` quantity:
+
+    >>> print(unit_converter.code_to_base("energy"))
+    0.043364104241800934 electron_volt
+
+    or as a scalar:
+
+    >>> print(unit_converter.code_to_base_value("energy"))
+    0.043364104241800934
+
+    Alternatively, the unit converter can also be used as decorators for functions that return an array scaled into
+    appropriate units:
+
+    >>> @unit_converter(quantity="energy", conversion="code_to_base")
+    >>>    def return_ones():
+    >>>        return np.ones(5)
+    >>> print(return_ones())
+    [0.0433641 0.0433641 0.0433641 0.0433641 0.0433641]
+
+    The decorator can also be used to assign units for numpy arrays
+    (for more info see https://pint.readthedocs.io/en/0.10.1/numpy.html)
+
+    >>> @unit_converter(quantity="energy", conversion="base_units")
+    >>> def return_ones_ev():
+    >>>     return np.ones(5)
+    >>> print(return_ones_ev())
+    [1.0 1.0 1.0 1.0 1.0] electron_volt
+
+    """
+
+    def __init__(self, base_units, code_units):
+        """
+
+        Args:
+            base_units (pyiron_base.generic.units.PyironUnitRegistry): Base unit registry
+            code_units (pyiron_base.generic.units.PyironUnitRegistry): Code specific unit registry
+        """
+        self._base_units = base_units
+        self._code_units = code_units
+
+    def code_to_base(self, quantity):
+        """
+        Get the conversion factor as a `pint` quantity from code to base units
+
+        Args:
+            quantity (str): Name of quantity
+
+        Returns:
+            pint.quantity.Quantity: Conversion factor as a `pint` quantity
+        """
+        return (1 * self._code_units[quantity]).to(self._base_units[quantity])
+
+    def base_to_code(self, quantity):
+        """
+        Get the conversion factor as a `pint` quantity from base to code units
+
+        Args:
+            quantity (str): Name of quantity
+
+        Returns:
+            pint.quantity.Quantity: Conversion factor as a `pint` quantity
+        """
+        return (1 * self._base_units[quantity]).to(self._code_units[quantity])
+
+    def code_to_base_value(self, quantity):
+        """
+        Get the conversion factor as a scalar from code to base units
+
+        Args:
+            quantity (str): Name of quantity
+
+        Returns:
+            float: Conversion factor as a float
+        """
+        return self.code_to_base(quantity).magnitude
+
+    def base_to_code_value(self, quantity):
+        """
+        Get the conversion factor as a scalar from base to code units
+
+        Args:
+            quantity (str): Name of quantity
+
+        Returns:
+            float: Conversion factor as a float
+        """
+        return self.base_to_code(quantity).magnitude
+
+    def __call__(self, conversion, quantity):
```

Comment:
It would be cool if instead of having to provide the right string to `conversion`, I could do something like

```python
@unitconverter.code_to_base(quantity='energy')
def foo(...):
    ...
```

That means making four decorators and having them as attributes on `UnitConverter`.  If that turns out to be too much work the current solution is perfectly fine.

Reply:
Thanks for adding this, love it!